OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnMovement.cs

```diff
@@ -45,17 +45,66 @@ protected override void Created(Actor self)
 			base.Created(self);
 		}
 
-		void ITick.Tick(Actor self)
+		// playAttempt == called from TraitEnabled or StartedMoving(Vertically), else called from Stopped* or TraitDisabled
+		// vertical == called from one of the two *Vertically methods (or TraitEnabled), else called from some other method
+		void UpdateCondition(Actor self, bool grantAttempt = true, bool vertical = false)
```

Comment:
The `MovementType` could then be passed in here instead of `vertical`

Reply:
There doesn't seem to be any code (except for the conditionManager check) shared between the `grantAttempt` true and false cases - best to have two methods for grant vs revoke.